backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/builder/vminfo/VmInfoBuildUtils.java

```diff
@@ -1267,9 +1268,13 @@ public boolean isBlockDomainPath(String path) {
 
     public List<VmNumaNode> getVmNumaNodes(VM vm) {
         int onlineCpus = VmCpuCountHelper.getDynamicNumOfCpu(vm);
-        int vcpus = FeatureSupported.supportedInConfig(ConfigValues.HotPlugCpuSupported, vm.getCompatibilityVersion(), vm.getClusterArch()) && !VmCpuCountHelper.isResizeAndPinPolicy(vm)?
-                VmCpuCountHelper.calcMaxVCpu(vm, vm.getCompatibilityVersion())
-                : onlineCpus;
+        int vcpus = onlineCpus;
+        if (vm.getCpuPinningPolicy() != CpuPinningPolicy.DEDICATED) {
+            vcpus = FeatureSupported.supportedInConfig(ConfigValues.HotPlugCpuSupported, vm.getCompatibilityVersion(),
```

Comment:
we're starting to have more mess in the code by patching it with this logic all around - can we say introduce a place that returns whether hot-plug CPU for a VM is supported that takes into account not only ConfigValues.HotPlugCpuSupported but also its pinning policy?